engine/src/main/java/org/terasology/world/generator/plugin/WorldGeneratorListenerLibrary.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.generator.plugin;
+
+import java.util.List;
+import java.util.Map;
+
+@FunctionalInterface
+public interface WorldGeneratorListenerLibrary {
```

Comment:
What sort of non-default WorldGeneratorListenerLibrary do you imagine there might be? Why have this as a separate interface?

Reply:
I guess I was cargo cult following the plugin library style. I've removed the interface and just made it a single class.